packages/astro/src/cli/check/index.ts

```diff
@@ -1,101 +1,360 @@
+			{
+				ignored: ['**/node_modules/**'],
+				ignoreInitial: true,
 			}
+		);
+
+		this.#watcher.on('add', (file) => {
+			this.#addDocument(file);
+			this.#filesCount += 1;
+			this.#checkForDiagnostics();
 		});
-	});
+		this.#watcher.on('change', (file) => {
+			this.#addDocument(file);
+			this.#checkForDiagnostics();
+		});
+		this.#watcher.on('unlink', (file) => {
+			this.#diagnosticsChecker.removeDocument(file);
+			this.#filesCount -= 1;
+			this.#checkForDiagnostics();
+		});
+	}
 
-	console.log(
-		[
-			bold(`Result (${filesCount} file${filesCount === 1 ? '' : 's'}): `),
-			bold(red(`${result.errors} ${result.errors === 1 ? 'error' : 'errors'}`)),
-			bold(yellow(`${result.warnings} ${result.warnings === 1 ? 'warning' : 'warnings'}`)),
-			dim(`${result.hints} ${result.hints === 1 ? 'hint' : 'hints'}\n`),
-		].join(`\n${dim('-')} `)
-	);
+	/**
+	 * Add a document to the diagnostics checker
+	 * @param filePath Path to the file
+	 */
+	#addDocument(filePath: string) {
+		const text = fs.readFileSync(filePath, 'utf-8');
+		this.#diagnosticsChecker.upsertDocument({
+			uri: pathToFileURL(filePath).toString(),
+			text,
+		});
+	}
+
+	/**
+	 * Logs the result of the various diagnostics
+	 *
+	 * @param {Readonly<DiagnosticResult>} result Result emitted by AstroChecker.#breakDownDiagnostics
+	 */
+	#logDiagnosticsSeverity(result: Readonly<DiagnosticResult>) {
+		info(
+			this.#logging,
+			'diagnostics',
+			[
+				bold(`Result (${this.#filesCount} file${this.#filesCount === 1 ? '' : 's'}): `),
+				bold(red(`${result.errors} ${result.errors === 1 ? 'error' : 'errors'}`)),
+				bold(yellow(`${result.warnings} ${result.warnings === 1 ? 'warning' : 'warnings'}`)),
+				dim(`${result.hints} ${result.hints === 1 ? 'hint' : 'hints'}\n`),
+			].join(`\n${dim('-')} `)
+		);
+	}
+
+	/**
+	 * It loops through all diagnostics and break down diagnostics that are errors, warnings or hints.
+	 * @param {Readonly<GetDiagnosticsResult[]>} diagnostics
+	 */
+	#breakDownDiagnostics(diagnostics: Readonly<GetDiagnosticsResult[]>): DiagnosticResult {
+		let result: DiagnosticResult = {
+			errors: 0,
+			warnings: 0,
+			hints: 0,
+		};
+
+		diagnostics.forEach((diag) => {
+			diag.diagnostics.forEach((d) => {
+				info(this.#logging, 'diagnostics', `\n ${printDiagnostic(diag.fileUri, diag.text, d)}`);
 
-	const exitCode = result.errors ? 1 : 0;
-	return exitCode;
+				switch (d.severity) {
+					case DiagnosticSeverity.Error: {
+						result.errors++;
+						break;
+					}
+					case DiagnosticSeverity.Warning: {
+						result.warnings++;
+						break;
+					}
+					case DiagnosticSeverity.Hint: {
+						result.hints++;
+						break;
+					}
+				}
+			});
+		});
+
+		return result;
+	}
 }
 
 /**
  * Open all Astro files in the given directory and return the number of files found.
+ * @param {URL} workspaceUri
+ * @param {string[]} filePathsToIgnore
+ * @param {AstroCheck} checker
```

Comment:
I think we can remove these too as they are already annotated in TypeScript below.